### PR TITLE
Fix CircleCI error

### LINKED ISSUE #3772: CircleCI build fails with `No module named 'numpy'`
CircleCI build fails with the following error log:

https://app.circleci.com/pipelines/github/mlflow/mlflow/2847/workflows/14f9de2f-e424-49cd-bd46-1b24d0677991/jobs/2859

```diff
Defaulting to user installation because normal site-packages is not writeable
Processing /home/circleci/project
Processing ./tests/resources/mlflow-test-plugin
Collecting matplotlib
  Downloading matplotlib-3.3.3-cp36-cp36m-manylinux1_x86_64.whl (11.6 MB)

Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in /usr/local/lib/python3.6/site-packages (from matplotlib->-r ./dev/large-requirements.txt (line 2)) (2.4.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: six>=1.10.0 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (1.15.0)
Collecting moto==1.3.16
  Downloading moto-1.3.16-py2.py3-none-any.whl (879 kB)

Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (2.10)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
Requirement already satisfied: zipp in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (3.4.0)
Requirement already satisfied: cryptography>=2.3.0 in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (3.2.1)
Requirement already satisfied: six>=1.10.0 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (1.15.0)
Collecting plotly
  Downloading plotly-4.13.0-py2.py3-none-any.whl (13.1 MB)

Requirement already satisfied: six>=1.10.0 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (1.15.0)
Collecting pytest==3.2.1
  Downloading pytest-3.2.1-py2.py3-none-any.whl (186 kB)

Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
Collecting pytest-cov==2.6.0
  Downloading pytest_cov-2.6.0-py2.py3-none-any.whl (14 kB)
Collecting pytest-localserver==0.5.0
  Downloading pytest-localserver-0.5.0.tar.gz (20 kB)
Collecting scipy
  Downloading scipy-1.5.4-cp36-cp36m-manylinux1_x86_64.whl (25.9 MB)

Collecting sphinx==2.2.1
  Downloading Sphinx-2.2.1-py3-none-any.whl (3.3 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
Collecting sphinx-click==2.3.0
  Downloading sphinx_click-2.3.0-py2.py3-none-any.whl (7.6 kB)
INFO: pip is looking at multiple versions of sphinx to determine which version is compatible with other requirements. This could take a while.
+ INFO: pip is looking at multiple versions of scipy to determine which version is compatible with other requirements. This could take a while.
Collecting scipy
  Downloading scipy-1.5.3-cp36-cp36m-manylinux1_x86_64.whl (25.9 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.5.2-cp36-cp36m-manylinux1_x86_64.whl (25.9 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.5.1-cp36-cp36m-manylinux1_x86_64.whl (25.9 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.5.0-cp36-cp36m-manylinux1_x86_64.whl (25.9 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.4.1-cp36-cp36m-manylinux1_x86_64.whl (26.1 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.4.0-cp36-cp36m-manylinux1_x86_64.whl (26.1 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.3.3-cp36-cp36m-manylinux1_x86_64.whl (25.2 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
INFO: pip is looking at multiple versions of sphinx to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of scipy to determine which version is compatible with other requirements. This could take a while.
  Downloading scipy-1.3.2-cp36-cp36m-manylinux1_x86_64.whl (25.2 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.3.1-cp36-cp36m-manylinux1_x86_64.whl (25.2 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.3.0-cp36-cp36m-manylinux1_x86_64.whl (25.2 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.2.3-cp36-cp36m-manylinux1_x86_64.whl (24.8 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.2.2-cp36-cp36m-manylinux1_x86_64.whl (24.8 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. If you want to abort this run, you can press Ctrl + C to do so. To improve how pip performs, tell us what happened here: https://pip.pypa.io/surveys/backtracking
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. If you want to abort this run, you can press Ctrl + C to do so. To improve how pip performs, tell us what happened here: https://pip.pypa.io/surveys/backtracking
  Downloading scipy-1.2.1-cp36-cp36m-manylinux1_x86_64.whl (24.8 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.2.0-cp36-cp36m-manylinux1_x86_64.whl (26.6 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.1.0-cp36-cp36m-manylinux1_x86_64.whl (31.2 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.0.1-cp36-cp36m-manylinux1_x86_64.whl (50.0 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-1.0.0-cp36-cp36m-manylinux1_x86_64.whl (50.0 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.19.1-cp36-cp36m-manylinux1_x86_64.whl (48.2 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.19.0-cp36-cp36m-manylinux1_x86_64.whl (48.2 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.18.1-cp36-cp36m-manylinux1_x86_64.whl (42.5 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.18.0.zip (14.6 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.17.1.zip (13.8 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.17.0.zip (13.8 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.16.1.zip (13.6 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.16.0.zip (19.3 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.15.1.zip (12.7 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.15.0.zip (12.7 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.14.1.zip (12.2 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.14.0.zip (11.5 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.13.3.zip (12.1 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.13.2.zip (11.4 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.13.1.zip (11.3 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
  Downloading scipy-0.13.0.zip (11.3 MB)

Requirement already satisfied: packaging in /usr/local/lib/python3.6/site-packages (from sphinx==2.2.1) (20.7)
Requirement already satisfied: requests>=2.17.3 in /usr/local/lib/python3.6/site-packages (from mlflow==1.12.2.dev0) (2.25.0)
Requirement already satisfied: setuptools in /usr/local/lib/python3.6/site-packages (from moto==1.3.16->-r ./dev/small-requirements.txt (line 7)) (50.3.2)
+  Downloading scipy-0.12.1.zip (10.2 MB)

    ERROR: Command errored out with exit status 1:
     command: /usr/local/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-rxh822ig/scipy_5d4e968b15d84b5593def9a27e7d16da/setup.py'"'"'; __file__='"'"'/tmp/pip-install-rxh822ig/scipy_5d4e968b15d84b5593def9a27e7d16da/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-gezcfqut
         cwd: /tmp/pip-install-rxh822ig/scipy_5d4e968b15d84b5593def9a27e7d16da/
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-rxh822ig/scipy_5d4e968b15d84b5593def9a27e7d16da/setup.py", line 165, in <module>
        setup_package()
      File "/tmp/pip-install-rxh822ig/scipy_5d4e968b15d84b5593def9a27e7d16da/setup.py", line 145, in setup_package
        from numpy.distutils.core import setup
    ModuleNotFoundError: No module named 'numpy'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

It looks like the new dependency resolver in pip 20.3 tries to determine which `scipy` version is compatible with other packages and hit a very old one ([0.12.1](https://pypi.org/project/scipy/0.12.1/)) which contains a bug.

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Fixes #3772 

## How is this patch tested?

(Details)

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -20,6 +20,8 @@ jobs:
             INSTALL_SMALL_PYTHON_DEPS: true
             INSTALL_LARGE_PYTHON_DEPS: true
           command: |
+            pip --version
```
Comment (harupy) [inline .circleci/config.yml:23]: This prints out the following on CircleCI:

```
pip 20.3 from /usr/local/lib/python3.6/site-packages/pip (python 3.6)
```

Comment (harupy) [reply .circleci/config.yml:23]: Screen Shot 2020-12-03 at 11 15 40

File: .circleci/config.yml [hunk 2]
```diff
@@ -20,6 +20,8 @@ jobs:
...
+            pip install --upgrade 'pip<20.3'
```
Comment (dbczumar) [inline .circleci/config.yml:24]: do we want `pip!=20.3` so that we pick up a newer pip once this problem is fixed?

Comment (harupy) [reply .circleci/config.yml:24]: makes sense. I'll make the change

Comment (harupy) [reply .circleci/config.yml]: done!

**Review by dbczumar — APPROVED** [review]
LGTM! Maybe can add an inline comment explaining why 20.3 is bad?

**Comment by harupy** [comment]
@dbczumar Added a comment!

**Pull request merged**